app/helpers/FileStorage/LocalStorage/LocalFileStorage.php

```diff
@@ -464,6 +467,39 @@ public function deleteOldFiles(string $glob, int $threshold): int
         return $deleted;
     }
 
+    public function deleteByFilter(callable $filter): int
+    {
+        $rootDirLen = strlen($this->rootDirectory) + 1; // plus 1 for '/';
+        $dirIterator = new RecursiveDirectoryIterator(
+            $this->rootDirectory,
+            FilesystemIterator::CURRENT_AS_PATHNAME | FilesystemIterator::SKIP_DOTS
+        );
+        $recursiveIterator = new RecursiveIteratorIterator($dirIterator);
+
+        // iterate over files and filter them
+        $toDelete = []; // we store the paths as a list first to avoid any iterator confusions
+        foreach ($recursiveIterator as $realPath) {
+            if (!Strings::startsWith($realPath, $this->rootDirectory)) {
```

Comment:
This can really happen?

Reply:
It really should not. However, I am not certain what will happen if links are involved, so just to be sure... (better fail immediately than venture beyond the realms of sanity)